common/src/main/java/com/microsoft/identity/common/internal/dto/PrimaryRefreshTokenRecord.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import lombok.EqualsAndHashCode;
+
+import static com.microsoft.identity.common.internal.dto.PrimaryRefreshTokenRecord.SerializedNames.*;
+
+@EqualsAndHashCode(callSuper = true, doNotUseGetters = true)
+public class PrimaryRefreshTokenRecord extends Credential {
+
+    public static class SerializedNames extends Credential.SerializedNames {
+        /**
+         * String of family id.
+         */
+        public static final String FAMILY_ID = "family_id";
+
+        /**
+         * String of session_key.
+         */
+        public static final String SESSION_KEY = "session_key";
+
+        /**
+         * String of session_key_rolling_date.
+         */
+        public static final String SESSION_KEY_ROLLING_DATE = "session_key_rolling_date";
+
+        /**
+         * String of prt_protocol_version.
+         */
+        public static final String PRT_PROTOCOL_VERSION = "prt_protocol_version";
+    }
+
+    /**
+     * 1st Party Application Family ID.
+     */
+    @SerializedName(FAMILY_ID)
+    private String mFamilyId;
+
+    /**
+     * PRT expiry time. This value is returned from the server as refresh_token_expires_in that
+     * should be added to local time.
+     */
+    @SerializedName(EXPIRES_ON)
+    private String mExpiresOn;
+
+    /**
+     * Session key of PRT.
+     */
+    @SerializedName(SESSION_KEY)
+    private String mSessionKey;
+
+    /**
+     * The version of the PRT Protocol being used, version 3 supports no device_id.
+     */
+    @SerializedName(PRT_PROTOCOL_VERSION)
+    private String mPrtProtocolVersion;
+
+    /**
+     * Session key expiry time. This value is determined by the client and set to 60 days after the
+     * session key was initally issued
+     */
+    @SerializedName(SESSION_KEY_ROLLING_DATE)
+    private String mSessionKeyRollingDate;
+
+    /**
+     * Gets the family_id.
+     *
+     * @return The family_id to get.
+     */
+    public String getFamilyId() {
+        return mFamilyId;
+    }
+
+    /**
+     * Sets the family_id.
+     *
+     * @param familyId The family_id to set.
+     */
+    public void setFamilyId(final String familyId) {
+        mFamilyId = familyId;
+    }
+
+    public boolean isExpired(final String expirationTimeEpochSeconds) {
+        return Long.parseLong(expirationTimeEpochSeconds) * 1000 < System.currentTimeMillis();
+    }
+
+    @Override
+    public boolean isExpired() {
+        return isExpired(getExpiresOn());
+    }
+
+    /**
+     * Gets the expires_on.
+     * @return The expires_on to get.
+     */
+    public String getExpiresOn() {
+        return mExpiresOn;
+    }
+
+    /**
+     * Sets the expires_on.
+     * @param expiresOn The expires_on to set.
+     */
+    public void setExpiresOn(String expiresOn) {
+        mExpiresOn = expiresOn;
+    }
+
+    /**
+     * Gets the session_key.
+     * @return The session_key to get.
+     */
+    public String getSessionKey() {
```

Comment:
Did we ever close on storing this value elsewhere?

/cc @AdamBJohnsonx @hieunguyenmsft

Reply:
As far as I'm aware of, no.  We should be able to abstract away whether this is a key or key identifier.

Reply:
Sorry @iambmelt I missed this notificatio :( In iOS the session_key is stored in keychain storage + in memory.

Reply:
Continuing this discussion below, with @AdamBJohnsonx